go/types/meta_sequence.go

```diff
@@ -339,3 +339,31 @@ func (es emptySequence) getCompositeChildSequence(start uint64, length uint64, h
 func (es emptySequence) isLeaf() bool {
 	return es.height == 1
 }
+
+// Invokes |cb| on all "uncommitted" ptree nodes reachable from |v| in
+// "bottom-up" order. It traverses across values, but will only traverse out of
+// the chunk containing |v| in order to follow uncommitted ptree nodes.
+func iterateUncommittedChildren(v Value, cb func(sv Value)) {
+	if _, ok := v.(*Type); ok {
+		// Types can be in-memory cyclic, but they cannot reference uncommitted
+		// nodes. Just avoid traversing into them.
+		return
+	}
+
+	if c, ok := v.(Collection); ok {
+		if ms, ok := c.sequence().(metaSequence); ok {
+			// Only traverse uncommitted children of meta sequences
+			for _, mt := range ms.tuples {
+				if mt.child != nil {
+					iterateUncommittedChildren(mt.child, cb)
+					cb(mt.child)
+				}
+			}
+			return
+		}
+	}
+
+	v.WalkValues(func(v Value) {
+		iterateUncommittedChildren(v, cb)
+	})
```

Comment:
It's surprising that you do this here at the end. The comment, and function name, lead me to believe this function is all about visiting uncommitted children in prolly trees, which you explicitly handle above. What's this here for? I'm sure it's to cover some necessary case, (maybe explicit Refs in collections or structs...?) but an example in a comment would help me understand.